server/ctools_direct_response.rb

```diff
@@ -91,14 +91,24 @@ def self.filter(assignment)
 
     logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: assignment: [#{assignment.inspect}]"
     calendarItem = assignment['calendarItem']
-    return nil if calendarItem.nil?
+
+    if calendarItem.nil?
+      logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: assignment: fail for nil calendarItem"
+      return nil
+    end
 
     calendarTimeLabelKey = calendarItem['calendarTimeLabelKey']
-    return nil if calendarTimeLabelKey.nil?
+    if calendarTimeLabelKey.nil?
+      logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: assignment: fail for nil calendarTimeLabelKey"
+      return nil
+    end
+
+    if 'assignment.due.date'.casecmp(calendarTimeLabelKey) != 0
+      logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: assignment: fail for bad key: [#{calendarTimeLabelKey}]"
+      return nil
+    end
 
-    return nil unless calendarTimeLabelKey == 'assignment.due.date'
-    
-    return assignment
+    assignment
```

Comment:
The value of the late expression in a method is used as the return value even without an explicit return statement.  I have been using this convention for consistency.